src/coreclr/src/inc/corprof.idl

```diff
@@ -2507,6 +2520,34 @@ interface ICorProfilerCallback9 : ICorProfilerCallback8
     HRESULT DynamicMethodUnloaded([in] FunctionID functionId);
 }
 
+[
+    object,
+    uuid(CEC5B60E-C69C-495F-87F6-84D28EE16FFB),
+    pointer_default(unique),
+    local
+]
+interface ICorProfilerCallback10 : ICorProfilerCallback9
+{
+    // This event is triggered whenever an EventPipe event is configured to be delivered.
+    //
+    // Documentation Note: All pointers are only valid during the callback
+
+    HRESULT EventPipeEventDelivered(
+        [in] EVENTPIPE_PROVIDER provider,
+        [in] DWORD eventId,
+        [in] DWORD eventVersion,
+        [in] ULONG cbMetadataBlob,
+        [in, size_is(cbMetadataBlob)] LPCBYTE metadataBlob,
+        [in] ULONG cbEventData,
+        [in, size_is(cbEventData)] LPCBYTE eventData,
+        [in] LPCGUID pActivityId,
+        [in] LPCGUID pRelatedActivityId,
+        [in] ThreadID eventThread,
+        [in] ULONG numStackFrames,
+        [in, length_is(numStackFrames)] UINT_PTR stackFrames[]);
+
+    HRESULT EventPipeProviderCreated([in] EVENTPIPE_PROVIDER provider);
```

Comment:
I initially added it because I wanted to test my sample profiler that parsed eventpipe metadata. Then I could start a session and just add every provider that existed and try to parse every event.

I left it because I can imagine profiler developers using it for finding out what providers are available. I don't have a production scenario in mind. I do think it matches the existing ICorProfiler philosophy of letting the profiler know when something is created.

Reply:
Sounds good to me (as long as we can resolve the lock issue) 👍